src/Symfony/Bundle/SecurityBundle/DependencyInjection/Compiler/MermaidDumperCommandPass.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\DependencyInjection\Compiler;
+
+use Symfony\Bundle\SecurityBundle\Command\SecurityRoleHierarchyDumpCommand;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * Registers the SecurityRoleHierarchyDumpCommand only if RoleHierarchy is available.
+ *
+ * @author Damien Fernandes <damien.fernandes24@gmail.com>
+ */
+class MermaidDumperCommandPass implements CompilerPassInterface
```

Comment:
the compiler pass should not have `Mermaid` in its name, as this is about the `SecurityRoleHierarchyDumpCommand`